### PR TITLE
Pulling Repos

### PR DESCRIPTION
**Pull request by casesandberg**

### CODE REVIEW

**Comment by casesandberg** [comment]
Dont merge, gonna switch to the RPC calls

**Comment by casesandberg** [comment]
Ok @hharnisc take a look, this should be ready to 🚀

File: frontend/src/store/index.js [hunk 1]
```diff
@@ -25,6 +28,7 @@ export default createStore(
         },
       }),
       logger,
+      thunk,
```
Comment (hharnisc) [inline frontend/src/store/index.js]: Do we need thunk?

File: frontend/src/App.js [hunk 2]
```diff
@@ -1,7 +1,45 @@
...
-  return <div>Payload</div>
+type Props = {
+  onLoad: ({ token: string }) => void,
+  repos: RepoList,
 }
 
-export default App
+export class App extends React.Component<Props> {
+  componentDidMount() {
+    getTokenFromUrl(window.location.search).then(token =>
```
Comment (hharnisc) [inline frontend/src/App.js]: Nice! This definitely works for now :smile:

File: frontend/src/store/index.js (hunk 1 above)
Comment (casesandberg) [reply frontend/src/store/index.js]: Thunk allows our actions to return functions. I use this quite a lot when composing actions together: https://github.com/payload-app/payload/pull/5/files#diff-48bb2e3e3e5708d8083b1cff1fff3949R40

Comment (hharnisc) [reply frontend/src/store/index.js]: Ah I see the pattern! What if you dispatched one action `@@init` or `APP_INIT` and caught the init action in middleware which could trigger those actions closer to the code? Then no need for thunk! 

Not that we shouldn't use it, I think that `thunk` adds complexity!

Comment (casesandberg) [reply frontend/src/store/index.js]: That makes sense for doing app init stuff, but I don't think it makes sense to do all action composition in different middlewares. 

My usage here isn't best practice. There should be something like a `fetchRepos` action that then dispatches both of those actions, it shouldn't be in the connect function there. That was just me being lazy.

Maybe we should chat about this at some point, might be a faster convo 😄

**Comment by casesandberg** [comment]
Ok @hharnisc, with this change I think we should be good! Feel free to merge.

**Comment by hharnisc** [comment]
:shipit:

**Pull request merged**